src/App.vue

```diff
@@ -5,14 +5,52 @@
 </template>
 
 <script>
-import { mapState } from 'vuex'
+import { mapState, mapGetters } from 'vuex'
+import { Watermark } from 'watermark-js-plus'
 
 export default {
   name: 'App',
+  data() {
+    return {
+      watermark: null
+    }
+  },
   computed: {
     ...mapState({
       isRouterAlive: state => state.common.isRouterAlive
+    }),
+    ...mapGetters({
+      currentUser: 'currentUser',
+      publicSettings: 'publicSettings'
     })
+  },
+  watch: {
+    currentUser: {
+      handler(newVal) {
+        this.createWatermark()
+      }
+    },
+    'publicSettings.SECURITY_WATERMARK_ENABLED': {
+      handler(newVal) {
+        this.createWatermark()
+      }
+    }
+  },
+  methods: {
+    createWatermark() {
+      if (this.currentUser?.username && this.publicSettings?.SECURITY_WATERMARK_ENABLED) {
+        this.watermark = new Watermark({
+          content: this.currentUser.username,
+          width: 200,
+          height: 200,
+          rotate: 45,
+          fontWeight: 'normal',
+          fontColor: 'rgba(128, 128, 128, 0.3)'
+        })
+
+        this.watermark.create()
+      }
+    }
   }
 }
 </script>
```

Comment:
Please provide the actual C# code that you want to compare between the different versions of JavaScript for detailed analysis